crates/polars-core/src/datatypes/any_value.rs

```diff
@@ -80,6 +80,7 @@ pub enum AnyValue<'a> {
     Enum(u32, &'a RevMapping, SyncPtr<Utf8ViewArray>),
     #[cfg(feature = "dtype-categorical")]
     EnumOwned(u32, Arc<RevMapping>, SyncPtr<Utf8ViewArray>),
+    // @TODO: This should be a ListArray<i64>, DataType instead of a Series
```

Comment:
I don't thinks so. It is a single value of the list. Which is a `Series` or better, zero or more elements of the inner dtype of the list. Since the inner dtype is not known statically, this is a `Series`.